### DISCUSSION TITLE
When to stop the training?

### DISCUSSION THREAD

**Discussion by DavidGerva** [explosion/spaCy#8613] [Help: Best practices]
The model I am training reaches an f_score of about 75 after 20000 steps. (I would like to know if this is a good value or not, on average, in your opinion).

I have never trained the model for more than about half a day.
Does it make sense to train a model for a very long time even if you don't get any improvement for several epochs? Or do you recommend training for a few days just to try?
Do you have any reading material I can study on this topic?

**Comment by polm** [comment]
Nobody knows! This is very much an area of active research. 

I would say the conventional wisdom is that if you don't see improvement over long enough your training is "done" and unlikely to improve; this is reflected in the `patience` setting in spaCy configs. On the other hand [recent research](https://mathai-iclr.github.io/papers/#:~:text=grokking%3A%20generalization%20beyond%20overfitting%20on%20small%20algorithmic%20datasets) describes a phenomenon dubbed "grokking" where models improve significantly after a long plateau.

> The model I am training reaches an f_score of about 75 after 20000 steps. (I would like to know if this is a good value or not, on average, in your opinion).

This completely depends on your problem. There are some problems where an f1 of 75 is awful and some where it's far beyond state of the art. Your best bet is to look for papers about similar problems, with similar evaluation metrics, and see how they do. [PapersWithCode](https://paperswithcode.com/) may help.

**Reply by DavidGerva** [reply]
Thank you, as always, @polm! I'll try to investigate further!
